### DISCUSSION TITLE
KeyValue not working as expected

### DISCUSSION THREAD

**Discussion by LemonPie22** [filamentphp/filament#5005] [Help] [answered]
I have a form with a KeyValue component `Forms\Components\KeyValue::make('custom_fields'),`

Initially it looks fine:
![Screenshot 2022-11-19 at 22 31 46](https://user-images.githubusercontent.com/107488019/202874091-0bc7bb5b-c4d9-47fa-9f25-6b9ed4fec0df.png)

Also when I add some values:
![Screenshot 2022-11-19 at 22 32 38](https://user-images.githubusercontent.com/107488019/202874372-35b9a78a-9793-4cfc-8b7e-c640d3044c55.png)

Which get stored in a `json` column in the database just fine:
![Screenshot 2022-11-19 at 22 42 22](https://user-images.githubusercontent.com/107488019/202874385-28ec7dc8-e70d-4481-8d5e-e93f1a77d903.png)

But then it goes wrong and the form field looks like this:
![Screenshot 2022-11-19 at 22 44 54](https://user-images.githubusercontent.com/107488019/202874430-4e0bcb39-1738-4c8b-b39f-8337317c1f50.png)

Is this a bug or am I doing something wrong?

**Comment by danharrin** [comment, marked as answer]
You're missing the `array` cast on your model for this property

**Reply by LemonPie22** [reply]
Thanks @danharrin it works fine now.
